projects/gateway2/api/v1alpha1/gateway_parameters_types.go

```diff
@@ -618,6 +635,85 @@ func (in *AiExtension) GetPorts() []*corev1.ContainerPort {
 	return in.Ports
 }
 
+func (in *AiExtension) GetStats() *AiExtensionStats {
+	if in == nil {
+		return nil
+	}
+	return in.Stats
+}
+
+type AiExtensionStats struct {
+	// Set of custom labels to be added to the request metrics.
+	// These will be added on each request which goes through the AI Extension.
+	// +optional
+	CustomLabels []*CustomLabels `json:"customLabels,omitempty"`
+}
+
+func (in *AiExtensionStats) GetCustomLabels() []*CustomLabels {
+	if in == nil {
+		return nil
+	}
+	return in.CustomLabels
+}
+
+type CustomLabels struct {
```

Comment:
should this be singular?
```suggestion
type CustomLabel struct {
```